codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -1167,6 +1182,7 @@ private void writeComplete(ChannelFuture future, ChannelPromise promise) {
                             outboundClosed = true;
                         }
                     }
+                    promise.setFailure(error);
```

Comment:
Note that this was moved to this position to be consistent with `firstWriteFailed` and the general behavior of AbstractChannel, namely that we transition the state reflected by `Channel.isOpen` and `Channel.isActive`, before failing the outbound buffer, which would normally contain this promise.